### PR TITLE
Update links, fix wording in headers and sidebar

### PR DESCRIPTION
**Pull request by artemkovalyov**
## Context

- More semantic link slugs for Google to like it.
- Removed VDM from there
- Somewhat updated wording

Next step:
- Break this material down into the base and advanced cases.
- Extract logically independent items

### CODE REVIEW

File: docs/java/features/odata/generate-client.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
-id: java-generate-odata-vmd-v2-v4
-title: Typed OData client generator for Java
+id: generate-typed-odata-v2-and-v4-client-for-java
+title: Generate a typed OData client for Java
 hide_title: false
 hide_table_of_contents: false
-sidebar_label: Generate a typed OData client for OData
+sidebar_label: Generate a typed OData client for Java
 description: This article describes how to leverage the OData Generator to obtain Java classes from a service definition. These classes can then be used to build type-safe OData requests.
```
Comment (sacnl) [inline docs/java/features/odata/generate-client.md:7]:
```suggestion
description: This article describes how to leverage the OData client generator to obtain Java classes from a service definition. These classes can then be used to build type-safe OData requests.
```

File: docs/java/features/odata/use-vdm.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ---
-id: use-generated-odata-vdm-v2-v4
-title: Use the typed OData client to consume OData Services
+id: use-typed-odata-v2-and-v4-client-for-java
+title: Typed client to consume OData Services for Java
 hide_title: false
 hide_table_of_contents: false
-sidebar_label: Consume OData Services
-description: This article describes how the SAP Cloud SDK for Java can be used to build and run OData requests.
+sidebar_label: Typesafe consumption of Odata services
```
Comment (sacnl) [inline docs/java/features/odata/use-vdm.md:6]:
```suggestion
sidebar_label: Type-safe consumption of OData services
```

File: docs/java/features/odata/use-vdm.md [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
+description: Use SAP Cloud SDK for Java to build build and run OData requests in a typesafe way.
```
Comment (sacnl) [inline docs/java/features/odata/use-vdm.md:7]:
```suggestion
description: Use SAP Cloud SDK for Java to build and run OData requests in a typesafe way.
```

**Comment by artemkovalyov** [comment]
@sacnl, thanks for suggestions. I blanded them in.

**Pull request merged**